pipelines/tekton/azureml-container-pipeline/model-upload/Makefile

```
@@ -0,0 +1,23 @@
+# Creates a PVC and a pod which mounts the PVC create
+# SIZE - size of the storage used in the PVC created
+# PVC - name of the PVC 
+.PHONY: create
+create:
+	@oc process -f local-model-pvc-template.yaml -p PVC=${PVC} -p SIZE=${SIZE} | oc create -f -
+	@oc process -f local-model-to-pvc-pod-template.yaml -p PVC=${PVC} | oc create -f -
+	@oc get pod -w local-model-to-pvc-pod
+
+# Uploads a file to a PVC   
+# MODEL_PATH - local path to model file to upload 
+# PVC - name of the PVC
+# NAME - name of the file when uploaded to the pod
+.PHONY: upload_model
+upload_model:
+	@oc exec local-model-to-pvc-pod -- mkdir workspace/${PVC}/model_dir/pipelines/models/ -p
+	@oc cp ${MODEL_PATH} local-model-to-pvc-pod:/workspace/${PVC}/model_dir/pipelines/models/${NAME}
+
+# deletes the pod created from the template
+# PVC - name of he PVC that is mounted to the pod
+.PHONY: clean
+clean:
+	@oc process -f local-model-to-pvc-pod.yaml -p PVC=${PVC} | oc delete -f -
```

Given that the pod name is not currently configurable in the template, this might be slightly simpler (and at that point, a dedicated `make` target might be unnecessary?):

```suggestion
	@oc delete pod local-model-to-pvc-pod
```

---

Sure I will do this, but on another note should the name of the pod be configurable?

---

> the Pod isn't actually important, and can be deleted once the model has been uploaded, is that right?

Yes the pod is just there to use `oc cp` and is not important

---

> maybe a generated name (or a name based on the PVC name, like ${PVC}-upload) 

Yes that would be good I think, but right now because they are in different targets that might be a pain but if we figure out that I would be happy to make that change

---

> recipe, rather than having them spread out in different places? 

This was what I wanted first, do it all in one make target. I couldn't figure out how to wait for the pod to be fully created before going on to the next command though (I didn't go too deep, so it is probably possible). 

If we can get something like `create pod, create PVC -> mount PVC -> oc cp -> delete pod` all in one make target that would be great

---

There's a `kubectl wait` subcommand, so I think what you're looking for is something like this, is it? I haven't tried it out:

```bash
@oc process -f local-model-to-pvc-pod-template.yaml -p PVC=${PVC} | oc create -f -
@oc wait pod local-model-to-pvc-pod --for condition=Ready=True --timeout=120s
@oc exec local-model-to-pvc-pod -- mkdir -p workspace/${PVC}/model_dir/pipelines/models/
@oc cp ${MODEL_PATH} local-model-to-pvc-pod:/workspace/${PVC}/model_dir/pipelines/models/${NAME}
@oc delete pod local-model-to-pvc-pod
```

---

Done 👍

---

Nice! Is the clean target still needed now, or could the pod be deleted immediately after the copy in the create target?

---

Marking this as unresolved as I still have this question above. :slightly_smiling_face: